secret_store/src/listener/service_contract.rs

```diff
@@ -181,16 +290,33 @@ impl ServiceContract for OnChainServiceContract {
 			from_block: BlockId::Hash(first_block),
 			to_block: BlockId::Hash(last_block),
 			address: Some(vec![address]),
-			topics: vec![
-				Some(vec![*SERVER_KEY_REQUESTED_EVENT_NAME_HASH]),
-				None,
-				None,
-				None,
-			],
+			topics: vec![Some(mask_topics(&self.mask))],
 			limit: None,
 		});
 
-		Box::new(request_logs.into_iter().map(|log| log.entry.topics))
+		Box::new(request_logs.into_iter()
+			.filter_map(|log| {
+				let raw_log: RawLog = (log.entry.topics.into_iter().map(|t| t.0.into()).collect(), log.entry.data).into();
+				if raw_log.topics[0] == *SERVER_KEY_GENERATION_REQUESTED_EVENT_NAME_HASH {
```

Comment:
Can we make this a match statement for better readability?

Reply:
Not sure :) These are not actually a constant expressions, but `Deref` calls (it is `lazy_static` 'consts'). I could replace with `match topics[0] { x if x == *SERVER_KEY... }`, but that's definitely not better :)